Sources/FluentKit/Query/QueryBuilder.swift

```diff
@@ -267,7 +276,18 @@ public final class QueryBuilder<Model>
     }
 
     @discardableResult
-    public func filter<Joined>(_ alias: Joined.Type, _ filter: ModelFilter<Joined>) -> Self
+    public func filter<Alias>(_ alias: Alias.Type, _ filter: ModelFieldFilter<Alias.Model, Alias.Model>) -> Self
+        where Alias: ModelAlias
+    {
+        return self.filter(
+            .field(path: filter.lhsPath, schema: Alias.alias, alias: nil),
+            filter.method,
+            .field(path: filter.rhsPath, schema: Alias.alias, alias: nil)
+        )
+    }
+
+    @discardableResult
+    public func filter<Joined>(_ alias: Joined.Type, _ filter: ModelValueFilter<Joined>) -> Self
```

Comment:
Not sure if I'm doing the right thing here, haven't really wrapped my head around `Joined` and `Alias`

Reply:
Yeah, this seems right :) Alias and Joined are basically exact the same except that Alias has a different table name. You can see an example of how that works in practice here: https://github.com/vapor/fluent-kit/blob/master/Sources/FluentBenchmark/FluentBenchmarker.swift#L1258